### DISCUSSION TITLE
How to get back an ISO 639-1 code

### DISCUSSION THREAD

**Discussion by kensanata** [pemistahl/lingua-go#50] [Q&A] [answered]
How do I get back the ISO 639-1 code for a language? My use case is a text that is turned to HTML and I want to set the `lang` attribute, e.g. `lang="en"` or `lang="de"`.

Sadly, I'm new to Go, so this is what I have in a test and I always get back EU for English.

```go
func TestLanguage(t *testing.T) {
	language := lingua.English
	if lingua.IsoCode639_1(language) != lingua.EN {
		t.Logf("Language: %s, ISO 639-1: %s", language.String(), lingua.IsoCode639_1(language).String())
		t.Fail()
	}
}
```

Result:

```text
    page_test.go:84: Language: English, ISO 639-1: EU
```

I guess I'm confused about the `IsoCode639_1` func and type?

**Comment by pemistahl** [comment, marked as answer]
Hi Alex, it's as simple as calling [`lingua.English.IsoCode639_1()`](https://pkg.go.dev/github.com/pemistahl/lingua-go#Language.IsoCode639_1). 

Conversely, if you want to get the language for a specific ISO code, just call [`lingua.GetLanguageFromIsoCode639_1(lingua.EN)`](https://pkg.go.dev/github.com/pemistahl/lingua-go#GetLanguageFromIsoCode639_1). While I'm thinking about it...it probably makes sense to implement `lingua.EN.Language()` for the next version. That would be much more elegant. Thank you for pointing me to this idea. (-:

**Reply by kensanata** [reply]
Thanks! That worked. :)
